### DISCUSSION TITLE
[design decision] Support for fractions

### DISCUSSION THREAD

**Discussion by josdejong** [tabular-json/tabular-json#3] [Ideas]
The `number` format of JSON is quite flexible and generally powerful enough. It supports notations like:

```
42
1.23
-2.3e4
2.3E+4
2.3e-4
```

In short, it supports integer numbers and floating point numbers. What it cannot represent effectively though is fractions, since floating point numbers can only represent a limited number of digits. For example `2/3` is represented as `0.6666666666666666` as a regular floating point number.

It may be worth implementing support for fractions, which can be written in two different ways:

1. As a fraction like `1/7`
2. As a repeating sequence like `0.(142857)`

I think though that the added value is relatively low, since this is not a common use case. And though it is not efficient, it is possible to write a fraction as precise as possible just using _a lot_ of digits.

Complexity: Low
Added value: Low

**Comment by gwhitney** [comment]
I think a real difficulty is that there is not a standard datatype in all of the target languages to represent exact fractions, to the best of my knowledge. On the other hand, at least JavaScript and Python support arbitrary-precision integers, so it seems as though Tabular-JSON should (maybe it does already, but I couldn't quite discern that from the intro page).

**Reply by josdejong** [reply]
That is true, most languages do not have fractions built-in. However, if there is no fraction data type available, you can easily fallback on evaluating the result of the fraction and returning the result as a floating point value.

**Comment by josdejong** [comment]
Let's not add support for fractions, I think it's too much of an edge case, complicating the standard but not adding much value for most people.
